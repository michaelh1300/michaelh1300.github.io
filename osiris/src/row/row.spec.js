import { shallowMount } from '@vue/test-utils';
import Row from './row.vue';

describe('Row', () => {
  let wrapper;

  describe('when pass no props', () => {
    beforeEach(() => {
      wrapper = shallowMount(Row);
    });

    it('should match with the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have expected classList', () => {
      const expected = [
        'o-row',
        'is-justify-flex-start',
        'is-align-flex-start',
        'is-content-flex-start',
        'is-direction-row',
        'is-wrap',
      ];
      expect(wrapper.vm.classList).toEqual(expected);
    });
  });

  describe('when pass some props', () => {
    const props = {
      justify: 'flex-end',
      align: 'flex-end',
      content: 'center',
      direction: 'column',
      wrap: 'no-wrap',
    };

    describe(`and the props are ${JSON.stringify(props)}`, () => {
      beforeEach(() => {
        wrapper = shallowMount(Row, {
          propsData: props,
        });
      });

      it('should match with the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

      it('should have an expected classList', () => {
        const expected = [
          'o-row',
          'is-justify-flex-end',
          'is-align-flex-end',
          'is-content-center',
          'is-direction-column',
          'is-no-wrap',
        ];

        expect(wrapper.vm.classList).toEqual(expected);
      });
    });

    describe('and the prop is another tag like section', () => {
      beforeEach(() => {
        wrapper = shallowMount(Row, {
          propsData: { tag: 'section' },
        });
      });

      it('should match with the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
});
