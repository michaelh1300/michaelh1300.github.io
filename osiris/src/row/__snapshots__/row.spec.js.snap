// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row when pass no props should match with the snapshot 1`] = `<div class="o-row is-justify-flex-start is-align-flex-start is-content-flex-start is-direction-row is-wrap" style="margin-left: -10px; margin-right: -10px;"></div>`;

exports[`Row when pass some props and the prop is another tag like section should match with the snapshot 1`] = `<section class="o-row is-justify-flex-start is-align-flex-start is-content-flex-start is-direction-row is-wrap" style="margin-left: -10px; margin-right: -10px;"></section>`;

exports[`Row when pass some props and the props are {"justify":"flex-end","align":"flex-end","content":"center","direction":"column","wrap":"no-wrap"} should match with the snapshot 1`] = `<div class="o-row is-justify-flex-end is-align-flex-end is-content-center is-direction-column is-no-wrap" style="margin-left: -10px; margin-right: -10px;"></div>`;
